src/main/groovy/com/ullink/NuGetPack.groovy

```diff
@@ -122,7 +123,8 @@ class NuGetPack extends BaseNuGet {
 
     File getPackageFile() {
         def spec = getNuspec()
-        def version = spec.metadata.version ?: project.version
-        new File(getDestinationDir(), spec.metadata.id.toString() + '.' + version + '.nupkg')
+        def version = spec?.metadata?.version ?: project.version
+        def id = spec?.metadata?.id?.toString() ?: FilenameUtils.removeExtension(project.msbuild.projects.find { csprojPath.equalsIgnoreCase(it.value.projectFile) }.value.dotnetAssemblyFile.name)
```

Comment:
if the msbuild task is not used, then the nuspec file should be provided, right? so it will use the former behaviour without requesting the msbuild task, I guess, no?